### PR TITLE
Integrate login api with frontend

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
## Description of Change

- Called login api from login hooks
- Updated state after login
- Set access token at local storage

## Associated Ticket Link(s)

- https://trello.com/c/m4rbZzoW

## Related Pull Request(s)

- N/A

### CODE REVIEW

File: modules/Login/hooks/useLoginFormData.ts [hunk 1]
```diff
@@ -1,12 +1,44 @@
...
 const useLoginFormData = () => {
-  const onSubmit = (data: TLoginFormData) => {
-    // TO DO: will remove it later after integrating api call
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const dispatch = useAppDispatch();
+  const [login] = useLoginMutation();
+
+  const onSubmit = async (data: TLoginFormData) => {
     console.log(data);
+    setIsSubmitting(true);
```
Comment (ishmam-iqbal-sazim) [inline modules/Login/hooks/useLoginFormData.ts:24]: Remove console.log above if unnecessary

File: shared/components/wrappers/AuthGuard/AuthGuard.utils.ts [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
-      return HOME_ROUTE;
+    case ERole.TEACHER:
+      // Will Add route after creating
+      return "/";
+    case ERole.STUDENT:
+      // Will add route after creting
+      return "/";
     default:
       return "/";
   }
```
Comment (ishmam-iqbal-sazim) [inline shared/components/wrappers/AuthGuard/AuthGuard.utils.ts:21]: JC - Is this method needed right now? If users are going to be routed to dashboard by default upon logging in or registering, will this ever be needed?

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left some comments

File: shared/components/wrappers/AuthGuard/AuthGuard.utils.ts (hunk 2 above)
Comment (faysal-alam-sazim) [reply shared/components/wrappers/AuthGuard/AuthGuard.utils.ts:21]: After login or registering user should redirect to their respective dashboard page. I think this method will be needed. But this redirecting can be done without this method also. As this is in the template I didn't want to remove it.

Comment (faysal-alam-sazim) [reply shared/components/wrappers/AuthGuard/AuthGuard.utils.ts]: Removed it along with hoc component

**Pull request merged**